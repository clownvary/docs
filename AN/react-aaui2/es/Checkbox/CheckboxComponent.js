import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _Object$defineProperties from 'babel-runtime/core-js/object/define-properties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { PureComponent } from 'react';
import { bool, string, object, func, node, any } from 'prop-types';
import classNames from 'classnames';

import { identity } from '../shared/utils';

var Checkbox = function (_PureComponent) {
  _inherits(Checkbox, _PureComponent);

  function Checkbox(props) {
    _classCallCheck(this, Checkbox);

    var _this = _possibleConstructorReturn(this, _PureComponent.call(this, props));

    _initialiseProps.call(_this);

    var checked = 'checked' in props ? props.checked : props.defaultChecked;

    _this.state = {
      checked: !!checked
    };
    return _this;
  }

  Checkbox.prototype.componentDidMount = function componentDidMount() {
    _Object$defineProperties(this, {
      checked: {
        get: function get() {
          return this.input.checked;
        },
        set: function set(v) {
          this.input.checked = !!v;
          this.setState({
            checked: !!v
          });
        }
      }
    });
  };

  Checkbox.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    if ('checked' in nextProps) {
      this.setState({
        checked: !!nextProps.checked
      });
    }
  };

  Checkbox.prototype.render = function render() {
    var _classNames;

    var _props = this.props,
        toggle = _props.toggle,
        size = _props.size,
        children = _props.children,
        className = _props.className,
        style = _props.style,
        disabled = _props.disabled,
        rest = _objectWithoutProperties(_props, ['toggle', 'size', 'children', 'className', 'style', 'disabled']);

    var checked = this.state.checked;

    var checkboxWrapperClasses = classNames({
      'checkbox-wrapper': true,
      toggle: toggle,
      'toggle--empty': !children && toggle
    }, className);
    var checkboxClasses = classNames((_classNames = {
      checkbox: !toggle,
      'checkbox--checked': !toggle && checked,
      'checkbox--disabled': !toggle && disabled
    }, _classNames['checkbox--' + size] = size, _classNames));

    return React.createElement(
      'label',
      { className: checkboxWrapperClasses, style: style },
      React.createElement(
        'span',
        { className: checkboxClasses },
        React.createElement('input', _extends({
          ref: this.setWrappedComponentInstance,
          className: 'checkbox__input',
          disabled: disabled,
          type: 'checkbox',
          'aria-hidden': true
        }, rest, {
          checked: checked,
          onChange: this.handleChange
        })),
        toggle ? React.createElement('span', { className: 'toggle__text' }) : React.createElement('span', {
          className: 'checkbox__inner',
          role: 'checkbox',
          'aria-disabled': disabled,
          'aria-checked': checked
        }),
        children && React.createElement(
          'span',
          { className: 'checkbox__text' },
          children
        )
      )
    );
  };

  return Checkbox;
}(PureComponent);

Checkbox.displayName = 'AUICheckbox';
Checkbox.propTypes = {
  disabled: bool,
  checked: bool,
  defaultChecked: bool,
  toggle: bool,
  size: string,
  className: string,
  style: object, // eslint-disable-line
  onChange: func,
  children: node,
  value: any // eslint-disable-line
};
Checkbox.defaultProps = {
  onChange: identity,
  defaultChecked: false
};

var _initialiseProps = function _initialiseProps() {
  var _this2 = this;

  this.setWrappedComponentInstance = function (input) {
    _this2.input = input;
  };

  this.triggerOnChange = function (e) {
    var onChange = _this2.props.onChange;

    var checked = e.target.checked;

    if (!('checked' in _this2.props)) {
      _this2.setState({
        checked: checked
      });
    }

    // If you want to access the event properties in an asynchronous way,
    // you should call `event.persist()`` on the event,
    // which will remove the synthetic event from the pool
    // and allow references to the event to be retained by user code.
    e.persist && e.persist();

    // The `value` prop is consumed by `CheckboxGroup`
    onChange(e, { value: _this2.props.value });
  };

  this.handleChange = function (e) {
    var disabled = _this2.props.disabled;


    if (disabled) {
      return;
    }

    _this2.triggerOnChange(e);
  };
};

export default Checkbox;