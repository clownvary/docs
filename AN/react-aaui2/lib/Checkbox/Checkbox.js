'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _defineProperties = require('babel-runtime/core-js/object/define-properties');

var _defineProperties2 = _interopRequireDefault(_defineProperties);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _CheckboxComponent = require('./CheckboxComponent');

var _CheckboxComponent2 = _interopRequireDefault(_CheckboxComponent);

var _utils = require('../shared/utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Checkbox = function (_Component) {
  (0, _inherits3.default)(Checkbox, _Component);

  function Checkbox() {
    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, Checkbox);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.setRef = function (checkboxComponent) {
      _this.checkboxComponent = checkboxComponent;
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  Checkbox.prototype.componentDidMount = function componentDidMount() {
    (0, _defineProperties2.default)(this, {
      checked: {
        get: function get() {
          return this.checkboxComponent.checked;
        },
        set: function set(v) {
          this.checkboxComponent.checked = !!v;
        }
      }
    });
  };

  Checkbox.prototype.render = function render() {
    var _props = this.props,
        children = _props.children,
        rest = (0, _objectWithoutProperties3.default)(_props, ['children']);
    var auiCheckboxGroup = this.context.auiCheckboxGroup;

    // Proxy `onChange`, `checked` and `disabled` to `auiCheckboxGroup`

    if (auiCheckboxGroup) {
      rest.onChange = auiCheckboxGroup.onChange;
      rest.checked = auiCheckboxGroup.value && auiCheckboxGroup.value.indexOf(this.props.value) !== -1;
      rest.disabled = this.props.disabled || auiCheckboxGroup.disabled;
      rest.size = this.props.size || auiCheckboxGroup.size;
      rest.name = this.props.name || auiCheckboxGroup.name;
    }

    return _react2.default.createElement(
      _CheckboxComponent2.default,
      (0, _extends3.default)({ ref: this.setRef }, rest),
      children
    );
  };

  return Checkbox;
}(_react.Component);

Checkbox.displayName = 'AUICheckbox';
Checkbox.propTypes = {
  disabled: _propTypes.bool,
  size: _propTypes.string,
  name: _propTypes.string,
  children: _propTypes.node,
  onChange: _propTypes.func,
  value: _propTypes.any // eslint-disable-line
};
Checkbox.defaultProps = {
  onChange: _utils.identity
};
Checkbox.contextTypes = {
  auiCheckboxGroup: _propTypes.object };
exports.default = Checkbox;
module.exports = exports['default'];