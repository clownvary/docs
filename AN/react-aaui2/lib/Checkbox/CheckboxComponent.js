'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _defineProperties = require('babel-runtime/core-js/object/define-properties');

var _defineProperties2 = _interopRequireDefault(_defineProperties);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _utils = require('../shared/utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Checkbox = function (_PureComponent) {
  (0, _inherits3.default)(Checkbox, _PureComponent);

  function Checkbox(props) {
    (0, _classCallCheck3.default)(this, Checkbox);

    var _this = (0, _possibleConstructorReturn3.default)(this, _PureComponent.call(this, props));

    _initialiseProps.call(_this);

    var checked = 'checked' in props ? props.checked : props.defaultChecked;

    _this.state = {
      checked: !!checked
    };
    return _this;
  }

  Checkbox.prototype.componentDidMount = function componentDidMount() {
    (0, _defineProperties2.default)(this, {
      checked: {
        get: function get() {
          return this.input.checked;
        },
        set: function set(v) {
          this.input.checked = !!v;
          this.setState({
            checked: !!v
          });
        }
      }
    });
  };

  Checkbox.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    if ('checked' in nextProps) {
      this.setState({
        checked: !!nextProps.checked
      });
    }
  };

  Checkbox.prototype.render = function render() {
    var _classNames;

    var _props = this.props,
        toggle = _props.toggle,
        size = _props.size,
        children = _props.children,
        className = _props.className,
        style = _props.style,
        disabled = _props.disabled,
        rest = (0, _objectWithoutProperties3.default)(_props, ['toggle', 'size', 'children', 'className', 'style', 'disabled']);
    var checked = this.state.checked;

    var checkboxWrapperClasses = (0, _classnames2.default)({
      'checkbox-wrapper': true,
      toggle: toggle,
      'toggle--empty': !children && toggle
    }, className);
    var checkboxClasses = (0, _classnames2.default)((_classNames = {
      checkbox: !toggle,
      'checkbox--checked': !toggle && checked,
      'checkbox--disabled': !toggle && disabled
    }, _classNames['checkbox--' + size] = size, _classNames));

    return _react2.default.createElement(
      'label',
      { className: checkboxWrapperClasses, style: style },
      _react2.default.createElement(
        'span',
        { className: checkboxClasses },
        _react2.default.createElement('input', (0, _extends3.default)({
          ref: this.setWrappedComponentInstance,
          className: 'checkbox__input',
          disabled: disabled,
          type: 'checkbox',
          'aria-hidden': true
        }, rest, {
          checked: checked,
          onChange: this.handleChange
        })),
        toggle ? _react2.default.createElement('span', { className: 'toggle__text' }) : _react2.default.createElement('span', {
          className: 'checkbox__inner',
          role: 'checkbox',
          'aria-disabled': disabled,
          'aria-checked': checked
        }),
        children && _react2.default.createElement(
          'span',
          { className: 'checkbox__text' },
          children
        )
      )
    );
  };

  return Checkbox;
}(_react.PureComponent);

Checkbox.displayName = 'AUICheckbox';
Checkbox.propTypes = {
  disabled: _propTypes.bool,
  checked: _propTypes.bool,
  defaultChecked: _propTypes.bool,
  toggle: _propTypes.bool,
  size: _propTypes.string,
  className: _propTypes.string,
  style: _propTypes.object, // eslint-disable-line
  onChange: _propTypes.func,
  children: _propTypes.node,
  value: _propTypes.any // eslint-disable-line
};
Checkbox.defaultProps = {
  onChange: _utils.identity,
  defaultChecked: false
};

var _initialiseProps = function _initialiseProps() {
  var _this2 = this;

  this.setWrappedComponentInstance = function (input) {
    _this2.input = input;
  };

  this.triggerOnChange = function (e) {
    var onChange = _this2.props.onChange;

    var checked = e.target.checked;

    if (!('checked' in _this2.props)) {
      _this2.setState({
        checked: checked
      });
    }

    // If you want to access the event properties in an asynchronous way,
    // you should call `event.persist()`` on the event,
    // which will remove the synthetic event from the pool
    // and allow references to the event to be retained by user code.
    e.persist && e.persist();

    // The `value` prop is consumed by `CheckboxGroup`
    onChange(e, { value: _this2.props.value });
  };

  this.handleChange = function (e) {
    var disabled = _this2.props.disabled;


    if (disabled) {
      return;
    }

    _this2.triggerOnChange(e);
  };
};

exports.default = Checkbox;
module.exports = exports['default'];