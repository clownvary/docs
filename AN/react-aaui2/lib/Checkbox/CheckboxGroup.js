'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Checkbox = require('./Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

var _utils = require('../shared/utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CheckboxGroup = function (_PureComponent) {
  (0, _inherits3.default)(CheckboxGroup, _PureComponent);

  function CheckboxGroup(props) {
    (0, _classCallCheck3.default)(this, CheckboxGroup);

    var _this = (0, _possibleConstructorReturn3.default)(this, _PureComponent.call(this, props));

    _initialiseProps.call(_this);

    var value = [];
    if ('value' in props) {
      value = props.value;
    } else if ('defaultValue' in props) {
      value = props.defaultValue;
    }

    _this.state = {
      value: value
    };
    return _this;
  }

  CheckboxGroup.prototype.getChildContext = function getChildContext() {
    return {
      auiCheckboxGroup: {
        onChange: this.onChange,
        value: this.state.value,
        disabled: this.props.disabled,
        size: this.props.size,
        name: this.props.name
      }
    };
  };

  CheckboxGroup.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    if ('value' in nextProps) {
      this.setState({
        value: nextProps.value
      });
    }
  };

  CheckboxGroup.prototype.render = function render() {
    var _props = this.props,
        className = _props.className,
        data = _props.data;

    var classes = (0, _classnames2.default)('checkbox-group', className);
    var children = this.props.children;

    if (data) {
      children = data.map(function (_ref, index) {
        var text = _ref.text,
            value = _ref.value,
            rest = (0, _objectWithoutProperties3.default)(_ref, ['text', 'value']);
        return _react2.default.createElement(
          _Checkbox2.default,
          (0, _extends3.default)({ key: index, value: value }, rest),
          text
        );
      });
    }

    return _react2.default.createElement(
      'div',
      { className: classes },
      children
    );
  };

  return CheckboxGroup;
}(_react.PureComponent);

CheckboxGroup.displayName = 'AUICheckboxGroup';
CheckboxGroup.propTypes = {
  disabled: _propTypes.bool,
  size: _propTypes.string,
  name: _propTypes.string,
  className: _propTypes.string,
  onChange: _propTypes.func,
  data: (0, _propTypes.arrayOf)((0, _propTypes.shape)({
    text: _propTypes.string.isRequired,
    value: _propTypes.any.isRequired // eslint-disable-line
  })),
  children: _propTypes.node,
  value: _propTypes.array, // eslint-disable-line
  defaultValue: _propTypes.array // eslint-disable-line
};
CheckboxGroup.defaultProps = {
  onChange: _utils.identity
};
CheckboxGroup.childContextTypes = {
  auiCheckboxGroup: _propTypes.object
};

var _initialiseProps = function _initialiseProps() {
  var _this2 = this;

  this.onChange = function (e, _ref2) {
    var optionValue = _ref2.value;

    // Should construct one new array for re-rendering
    var value = [].concat(_this2.state.value);
    var onChange = _this2.props.onChange;

    var index = value.indexOf(optionValue);

    if (index === -1) {
      // If NOT found, then add it
      value.push(optionValue);
    } else {
      // If found, then delete it
      value.splice(index, 1);
    }

    if (!('value' in _this2.props)) {
      _this2.setState({
        value: value
      });
    }

    onChange(value);
  };
};

exports.default = CheckboxGroup;
module.exports = exports['default'];