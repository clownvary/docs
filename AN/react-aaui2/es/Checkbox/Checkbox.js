import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _Object$defineProperties from 'babel-runtime/core-js/object/define-properties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Component } from 'react';
import { bool, string, object, func, node, any } from 'prop-types';

import CheckboxComponent from './CheckboxComponent';
import { identity } from '../shared/utils';

var Checkbox = function (_Component) {
  _inherits(Checkbox, _Component);

  function Checkbox() {
    var _temp, _this, _ret;

    _classCallCheck(this, Checkbox);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.setRef = function (checkboxComponent) {
      _this.checkboxComponent = checkboxComponent;
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  Checkbox.prototype.componentDidMount = function componentDidMount() {
    _Object$defineProperties(this, {
      checked: {
        get: function get() {
          return this.checkboxComponent.checked;
        },
        set: function set(v) {
          this.checkboxComponent.checked = !!v;
        }
      }
    });
  };

  Checkbox.prototype.render = function render() {
    var _props = this.props,
        children = _props.children,
        rest = _objectWithoutProperties(_props, ['children']);

    var auiCheckboxGroup = this.context.auiCheckboxGroup;

    // Proxy `onChange`, `checked` and `disabled` to `auiCheckboxGroup`

    if (auiCheckboxGroup) {
      rest.onChange = auiCheckboxGroup.onChange;
      rest.checked = auiCheckboxGroup.value && auiCheckboxGroup.value.indexOf(this.props.value) !== -1;
      rest.disabled = this.props.disabled || auiCheckboxGroup.disabled;
      rest.size = this.props.size || auiCheckboxGroup.size;
      rest.name = this.props.name || auiCheckboxGroup.name;
    }

    return React.createElement(
      CheckboxComponent,
      _extends({ ref: this.setRef }, rest),
      children
    );
  };

  return Checkbox;
}(Component);

Checkbox.displayName = 'AUICheckbox';
Checkbox.propTypes = {
  disabled: bool,
  size: string,
  name: string,
  children: node,
  onChange: func,
  value: any // eslint-disable-line
};
Checkbox.defaultProps = {
  onChange: identity
};
Checkbox.contextTypes = {
  auiCheckboxGroup: object };
export default Checkbox;