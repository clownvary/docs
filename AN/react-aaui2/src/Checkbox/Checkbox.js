import React, { Component } from 'react'
import { bool, string, object, func, node, any } from 'prop-types'

import CheckboxComponent from './CheckboxComponent'
import { identity } from '../shared/utils'

export default class Checkbox extends Component {
  static displayName = 'AUICheckbox'

  static propTypes = {
    disabled: bool,
    size: string,
    name: string,
    children: node,
    onChange: func,
    value: any, // eslint-disable-line
  }

  static defaultProps = {
    onChange: identity,
  }

  static contextTypes = {
    auiCheckboxGroup: object,
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  componentDidMount() {
    Object.defineProperties(this, {
      checked: {
        get() {
          return this.checkboxComponent.checked
        },
        set(v) {
          this.checkboxComponent.checked = !!v
        },
      },
    })
  }

  setRef = checkboxComponent => {
    this.checkboxComponent = checkboxComponent
  }

  render() {
    const { children, ...rest } = this.props
    const { auiCheckboxGroup } = this.context

    // Proxy `onChange`, `checked` and `disabled` to `auiCheckboxGroup`
    if (auiCheckboxGroup) {
      rest.onChange = auiCheckboxGroup.onChange
      rest.checked =
        auiCheckboxGroup.value &&
        auiCheckboxGroup.value.indexOf(this.props.value) !== -1
      rest.disabled = this.props.disabled || auiCheckboxGroup.disabled
      rest.size = this.props.size || auiCheckboxGroup.size
      rest.name = this.props.name || auiCheckboxGroup.name
    }

    return (
      <CheckboxComponent ref={this.setRef} {...rest}>
        {children}
      </CheckboxComponent>
    )
  }
}
