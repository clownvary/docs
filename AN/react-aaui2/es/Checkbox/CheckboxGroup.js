import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { PureComponent } from 'react';
import { bool, string, func, object, node, array, arrayOf, shape, any } from 'prop-types';
import classNames from 'classnames';

import Checkbox from './Checkbox';
import { identity } from '../shared/utils';

var CheckboxGroup = function (_PureComponent) {
  _inherits(CheckboxGroup, _PureComponent);

  function CheckboxGroup(props) {
    _classCallCheck(this, CheckboxGroup);

    var _this = _possibleConstructorReturn(this, _PureComponent.call(this, props));

    _initialiseProps.call(_this);

    var value = [];
    if ('value' in props) {
      value = props.value;
    } else if ('defaultValue' in props) {
      value = props.defaultValue;
    }

    _this.state = {
      value: value
    };
    return _this;
  }

  CheckboxGroup.prototype.getChildContext = function getChildContext() {
    return {
      auiCheckboxGroup: {
        onChange: this.onChange,
        value: this.state.value,
        disabled: this.props.disabled,
        size: this.props.size,
        name: this.props.name
      }
    };
  };

  CheckboxGroup.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    if ('value' in nextProps) {
      this.setState({
        value: nextProps.value
      });
    }
  };

  CheckboxGroup.prototype.render = function render() {
    var _props = this.props,
        className = _props.className,
        data = _props.data;

    var classes = classNames('checkbox-group', className);
    var children = this.props.children;

    if (data) {
      children = data.map(function (_ref, index) {
        var text = _ref.text,
            value = _ref.value,
            rest = _objectWithoutProperties(_ref, ['text', 'value']);

        return React.createElement(
          Checkbox,
          _extends({ key: index, value: value }, rest),
          text
        );
      });
    }

    return React.createElement(
      'div',
      { className: classes },
      children
    );
  };

  return CheckboxGroup;
}(PureComponent);

CheckboxGroup.displayName = 'AUICheckboxGroup';
CheckboxGroup.propTypes = {
  disabled: bool,
  size: string,
  name: string,
  className: string,
  onChange: func,
  data: arrayOf(shape({
    text: string.isRequired,
    value: any.isRequired // eslint-disable-line
  })),
  children: node,
  value: array, // eslint-disable-line
  defaultValue: array // eslint-disable-line
};
CheckboxGroup.defaultProps = {
  onChange: identity
};
CheckboxGroup.childContextTypes = {
  auiCheckboxGroup: object
};

var _initialiseProps = function _initialiseProps() {
  var _this2 = this;

  this.onChange = function (e, _ref2) {
    var optionValue = _ref2.value;

    // Should construct one new array for re-rendering
    var value = [].concat(_this2.state.value);
    var onChange = _this2.props.onChange;

    var index = value.indexOf(optionValue);

    if (index === -1) {
      // If NOT found, then add it
      value.push(optionValue);
    } else {
      // If found, then delete it
      value.splice(index, 1);
    }

    if (!('value' in _this2.props)) {
      _this2.setState({
        value: value
      });
    }

    onChange(value);
  };
};

export default CheckboxGroup;