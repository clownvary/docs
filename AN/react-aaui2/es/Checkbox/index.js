import Checkbox from './Checkbox';
import CheckboxGroup from './CheckboxGroup';

Checkbox.Group = CheckboxGroup;

export { Checkbox, CheckboxGroup };
export default Checkbox;