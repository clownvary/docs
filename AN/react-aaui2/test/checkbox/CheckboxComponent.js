import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'

import Checkbox from 'Checkbox/CheckboxComponent'

it('should render without errors', () => {
  const wrapper = shallow(<Checkbox />)

  expect(wrapper).toBeTruthy()
})

it('should render right thing', () => {
  const wrapper = shallow(<Checkbox>Checkbox</Checkbox>)

  expect(wrapper.find('label.checkbox-wrapper').length === 1).toBe(true)
  expect(wrapper.find('input').length === 1).toBe(true)
  expect(wrapper.find('span.checkbox__text').length === 1).toBe(true)
})

it('should render right thing when it is a toggle', () => {
  const wrapper = shallow(<Checkbox toggle />)

  expect(wrapper.find('label.toggle').length === 1).toBe(true)
  expect(wrapper.find('input').length === 1).toBe(true)
  expect(wrapper.find('span.toggle__text').length === 1).toBe(true)
})

it("should render right thing and don't change unexpected", () => {
  const tree = renderer.create(<Checkbox />).toJSON()

  expect(tree).toMatchSnapshot()
})

it('has tabIndex=-1 when disabled', () => {
  const wrapper = shallow(<Checkbox tabIndex={0} disabled />)

  expect(wrapper.find('[tabIndex=0]').length).toBe(1)
})

it('should has .checkbox by default', () => {
  const wrapper = shallow(<Checkbox>checkbox child</Checkbox>)

  expect(wrapper.hasClass('checkbox-wrapper')).toBe(true)
})

it('should has .checkbox and checkbox--sm for small sized checkbox', () => {
  const wrapper = shallow(<Checkbox size="sm" />)

  expect(wrapper.hasClass('checkbox-wrapper')).toBe(true)
  expect(wrapper.find('.checkbox').hasClass('checkbox--sm')).toBe(true)
})

it('should could checked through the instance property', () => {
  const wrapper = mount(<Checkbox defaultChecked />)

  expect(wrapper.instance().checked).toBe(true)
})

it('should could set checked through the instance `checked` property when defaultChecked', () => {
  const wrapper = mount(<Checkbox defaultChecked />)

  expect(wrapper.instance().checked).toBe(true)
  expect(wrapper.state('checked')).toBe(true)

  wrapper.instance().checked = false

  expect(wrapper.instance().checked).toBe(false)
  expect(wrapper.instance().input.checked).toBe(false)
  expect(wrapper.state('checked')).toBe(false)
})

it('should could set checked through the instance `checked` property when checked', () => {
  const wrapper = mount(<Checkbox checked />)

  expect(wrapper.instance().checked).toBe(true)
  expect(wrapper.state('checked')).toBe(true)

  wrapper.instance().checked = false

  expect(wrapper.instance().checked).toBe(false)
  expect(wrapper.instance().input.checked).toBe(false)

  expect(wrapper.state('checked')).toBe(false)
})

it('should could set checked through the instance `checked` property when unchecked', () => {
  const wrapper = mount(<Checkbox />)

  expect(wrapper.instance().checked).toBe(false)
  expect(wrapper.state('checked')).toBe(false)

  wrapper.instance().checked = true

  expect(wrapper.instance().checked).toBe(true)
  expect(wrapper.instance().input.checked).toBe(true)
  expect(wrapper.state('checked')).toBe(true)
})

it('should NOT call onChange when setting checked through the instance property', () => {
  const onChangeMock = jest.fn()
  const wrapper = mount(<Checkbox defaultChecked onChange={onChangeMock} />)

  wrapper.instance().checked = false

  expect(wrapper.instance().checked).toBe(false)
  expect(onChangeMock).not.toBeCalled()
})

it('should could sync props => state', () => {
  const wrapper = shallow(<Checkbox defaultChecked />)

  expect(wrapper.state('checked')).toBe(true)

  wrapper.setProps({
    useless: !wrapper.state('checked'),
  })

  wrapper.setProps({
    checked: !wrapper.state('checked'),
  })

  expect(wrapper.state('checked')).toBe(false)
})

it('should call onChange when triggering change event for the wrapped input', () => {
  const onChangeMock = jest.fn()
  const wrapper = shallow(<Checkbox onChange={onChangeMock} />)

  expect(wrapper.state('checked')).toBe(false)

  wrapper.find('input[type="checkbox"]').simulate('change', {
    persist: () => {},
    preventDefault: () => {},
    target: {
      checked: true,
    },
  })

  expect(onChangeMock).toBeCalled()
  expect(wrapper.state('checked')).toBe(true)
})

it("don't handle `onChange` event when setting `disabled` as true", () => {
  const onChangeMock = jest.fn()
  const wrapper = shallow(<Checkbox disabled checked onChange={onChangeMock} />)

  wrapper.find('input').simulate('change', {
    target: {
      value: false,
    },
  })

  expect(onChangeMock).not.toBeCalled()
})

it('should could call `stopPropagation` & `preventDefault`', () => {
  const stopPropagationMock = jest.fn()
  const preventDefaultMock = jest.fn()
  const onChangeMock = jest.fn(e => {
    e.stopPropagation()
    e.preventDefault()
  })
  const wrapper = shallow(<Checkbox onChange={onChangeMock} />)

  wrapper.find('input').simulate('change', {
    target: {
      checked: true,
    },
    stopPropagation: stopPropagationMock,
    preventDefault: preventDefaultMock,
  })

  expect(stopPropagationMock).toBeCalled()
  expect(preventDefaultMock).toBeCalled()
})
