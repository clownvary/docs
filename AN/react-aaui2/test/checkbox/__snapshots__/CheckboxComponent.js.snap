// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right thing and don't change unexpected 1`] = `
<label
  className="checkbox-wrapper"
  style={undefined}
>
  <span
    className="checkbox"
  >
    <input
      aria-hidden={true}
      checked={false}
      className="checkbox__input"
      defaultChecked={false}
      disabled={undefined}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      aria-checked={false}
      aria-disabled={undefined}
      className="checkbox__inner"
      role="checkbox"
    />
  </span>
</label>
`;
